_RELEASE/Packs/cube/Scripts/utils.lua

```diff
@@ -19,3 +132,381 @@ function clamp(input, min_val, max_val)
+function printTable(luaTable, whitespace)
+	whitespace = whitespace or "";
+	
+	if (type(luaTable) ~= "table") then return end;
+	
+	for k, v in pairs(luaTable) do
+		print(whitespace..tostring(k), v);
+		if (type(v) == "table" and v ~= luaTable) then
+			tablePrint(v, whitespace.."\t");
+		end
+	end
+end 
+
+-- Returns a random value between minimum and maximum, but is a floating point
+-- number.
+function randomFloat(minimum, maximum)
+	return math.random() * (maximum - minimum) + minimum;
+end
+
+-- Takes in a number value and follows true rounding rules.
+-- Decimals < .5 will round down, while >= .5 will round up.
+function roundFloat(floatingNumber)
+	return math.floor(floatingNumber + 0.5);
+end
+
+-- "Shuffles" an array by swapping elements randomly across a table.
+function shuffle(t)
+	for i = #t, 3, -1 do
+		local j = math.random(i - 1)
+		t[i], t[j] = t[j], t[i]
+	end
+end
+
+-- Takes in a number value and rounds the number to places decimal 
+-- places. Very useful to avoid floating point error.
+function simplifyFloat(number, places) 
+	return roundFloat(number * (10 ^ places)) / (10 ^ places);
+end
+
+--[[ EASING FUNCTIONS ]]--
+-- These are easing functions, which are a special subset of math functions that help with specific transitions.
+-- These output values from 0 to 1, and work best when paired with linear interpolation (lerp).
+-- All formulas are referenced from easings.net.
+
+-- Sine
+function easeInSine(x)
+	x = clamp(x, 0, 1);
+	return 1 - math.cos((x * math.pi) / 2);
+end
+
+function easeInOutSine(x)
+	x = clamp(x, 0, 1);
+	return -(math.cos(math.pi * x) - 1) / 2;
+end
+
+function easeOutSine(x)
+	x = clamp(x, 0, 1);
+	return math.sin((x * math.pi) / 2);
+end
+
+-- Quadratic
+function easeInQuad(x)
+	x = clamp(x, 0, 1);
+	return x ^ 2;
+end
+
+function easeInOutQuad(x) 
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 2 * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 2 / 2;
+end
+
+function easeOutQuad(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) * (1 - x);
+end
+
+-- Cubic
+function easeInCubic(x)
+	x = clamp(x, 0, 1);
+	return x ^ 3;
+end
+
+function easeInOutCubic(x)
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 4 * x * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 3 / 2;
+end
+
+function easeOutCubic(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) ^ 3;
+end
+
+-- Quartic
+function easeInQuart(x)
+	x = clamp(x, 0, 1);
+	return x ^ 4;
+end
+
+function easeInOutQuart(x)
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 8 * x * x * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 4 / 2; 
+end
+
+function easeOutQuart(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) ^ 4;
+end
+
+-- Quintic
+function easeInQuint(x)
+	x = clamp(x, 0, 1);
+	return x ^ 4;
+end
+
+function easeInOutQuint(x)
+	x = clamp(x, 0, 1);
+	if (x < 0.5) then
+		return 16 * x * x * x * x * x;
+	end
+	return 1 - (-2 * x + 2) ^ 5 / 2;
+end
+
+function easeOutQuint(x)
+	x = clamp(x, 0, 1);
+	return 1 - (1 - x) ^ 5;
+end
+
```

Comment:
Since we're doing polynomials, I thought why not add a general case? Not sure how useful it's going to end up being, but perhaps someone will want the fine tuning.

```suggestion
-- Polynomial degree e, pick any e ≥ 1
-- Linear, Quadratic, Cubic, Quartic, Quintic are special cases for e = 1, 2, 3, 4, 5 respectively
function easeInPoly(x, e)
	x = clamp(x, 0, 1);
	return x ^ e;
end

function easeInOutPoly(x, e)
	x = clamp(x, 0, 1);
	if (x < 0.5) then
		return (2 * x) ^ e / 2;
	end
	return 1 - (-2 * x + 2) ^ e / 2;
end

function easeOutPoly(x, e)
	x = clamp(x, 0, 1);
	return 1 - (1 - x) ^ e;
end

```